### PR TITLE
Update OpenSSH to 8.1

### PR DESCRIPTION
**Pull request by Mno-hime**
**Release notes**
* 8.0: https://www.openssh.com/txt/release-8.0
* 8.1: https://www.openssh.com/txt/release-8.1

Sync with OmniOS.

**Testing**
- successfully connected to 8.0 (itself), 7.9 (Fedora & openSUSE), and 7.4 (Debian 9.3)
- test suite passed `test_*` tests, the rest of the tests fail with (with `pfexec` or not, see https://github.com/omniosorg/omnios-build/blob/master/build/openssh/build.sh#L115):
```
run test connect.sh ...
ssh connect with failed
failed simple connect
```

### CODE REVIEW

File: components/network/openssh/patches/0008-Add-DisableBanner-option.patch [hunk 1]
```diff
@@ -137,20 +137,21 @@ diff -wpruN '--exclude=*.orig' a~/sshconnect2.c a/sshconnect2.c
...
++	char *msg = NULL, *raw;
+ 	size_t len;
+ 	int r;
  
  	debug3("%s", __func__);
--	msg = packet_get_string(&len);
-+	raw = packet_get_string(&len);
- 	lang = packet_get_string(NULL);
+-	if ((r = sshpkt_get_cstring(ssh, &msg, &len)) != 0 ||
++	if ((r = sshpkt_get_cstring(ssh, &raw, &len)) != 0 ||
```
Comment (Mno-hime) [inline components/network/openssh/patches/0008-Add-DisableBanner-option.patch]: This change ought to be double checked by someone.

**Comment by Mno-hime** [comment]
Omni had to [re-introduce GSS API key exchange](https://github.com/omniosorg/omnios-build/pull/1534) after removing it in 8.0.

File: components/network/openssh/patches/0033-test.patch [hunk 2]
```diff
@@ -0,0 +1,8 @@
+diff -wpruN '--exclude=*.orig' a~/regress/cfgparse.sh a/regress/cfgparse.sh
+--- a~/regress/cfgparse.sh	1970-01-01 00:00:00
++++ a/regress/cfgparse.sh	1970-01-01 00:00:00
+@@ -73,3 +73,4 @@ EOD
+ 
+ # cleanup
+ rm -f $OBJ/sshd_config.[012]
++exit 0
```
Comment (pyhalov) [inline components/network/openssh/patches/0033-test.patch]: Why is it needed? Please, update components/network/openssh/README if necessary.

Comment (Mno-hime) [reply components/network/openssh/patches/0033-test.patch]: It's not really clear why it's needed, I dropped it.

**Comment by pyhalov** [comment]
Please, restore removed patch comments, otherwise feel free to merge if it works for you.

**Comment by Mno-hime** [comment]
> Please, restore removed patch comments, otherwise feel free to merge if it works for you.

@pyhalov What you mean by "removed patch comments"? Can you clarify?

**Comment by pyhalov** [comment]
Some patches, for example,  components/network/openssh/patches/0016-PAM-setcred-failures.patch , contained comments or commit messages on the top, which were (accidentally) removed. Please, look through full changeset and restore them.

**Comment by Mno-hime** [comment]
I see, thanks for pointing that out. I brought back several of such removals.

**Comment by pyhalov** [comment]
Feel free to merge, if it works for you.

**Comment by alarcher** [comment]
@Mno-hime Should we merge this one as well?

**Pull request merged**